File: src/CSharp/893.NumSpecialEquivGroups.cs

using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.NumSpecialEquivGroups
{
public class Solution {
    public int NumSpecialEquivGroups(string[] A) {
        HashSet<string> hash = new HashSet<string>();
		foreach(var ch in A){
		    int len2 = ch.Length / 2;
		    int len1 = ch.Length - len2;
            char[] ch1 = new char[len1];
            char[] ch2 = new char[len2];
		    for (int i = 0; i + i < ch.Length; i++)
			    ch1[i] = ch[i + i];
		    for (int i = 0; i + i + 1 < ch.Length; i++)
			    ch2[i] = ch[i + i + 1];
		    Array.Sort(ch1);
            Array.Sort(ch2);
		    hash.Add(new string(ch1) + new string(ch2));
		}
		return hash.Count;
    }
}
}

File: src/Java/801-900/893.NumSpecialEquivGroups.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Set;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int numSpecialEquivGroups(String[] A) {
        HashSet<String> hash = new HashSet<>();
        for (String a : A){
            char[] builder1;
            char[] builder2;
            if (a.length() % 2 == 0){
                builder1 = new char[a.length() / 2];
                builder2 = new char[a.length() / 2];
            }
            else{
                builder1 = new char[a.length() / 2 + 1];
                builder2 = new char[a.length() / 2];
            }
            for (int i = 0; i < a.length(); i+=2) {
                builder1[i / 2] = a.charAt(i);
            }
            for (int i = 1; i < a.length(); i+=2) {
                builder2[i / 2] = a.charAt(i);
            }
            Arrays.sort(builder1);
            Arrays.sort(builder2);
            hash.add(String.valueOf(builder1) + String.valueOf(builder2));
        }
        return hash.size();
    }
}
